examples/automm/dataset.py

```diff
@@ -0,0 +1,490 @@
+    @property
+    def label_type(self):
+        return NUMERICAL
+    
+    @property
+    def metric(self):
+        return RMSE
+    
+    @property
+    def problem_type(self):
+        return REGRESSION
+
+
+class YahooTabularDataset(BaseTabularDataset):
+    _INFO = {
+        'train': {
+            'url': 's3://autogluon/datasets/tabular/yahoo/train.csv',
+            'sha1sum': '4a6ca419807da6991f755d0a349bfbac71cb6289'
+        },
+        'val': {
+            'url': 's3://autogluon/datasets/tabular/yahoo/val.csv',
+            'sha1sum': '50b5017d7aa01b0c07f0bc2e6b8c2ee298b2a1e2'
+        },
+        'test': {
+            'url': 's3://autogluon/datasets/tabular/yahoo/test.csv',
+            'sha1sum': '8b550c1dec3d1d6f980f7e51f3358336cd9d1973'
+        },
+    }
+    def __init__(self, split='train', path='./dataset/'):
+        assert split in ['train', 'val', 'test'], f'Unsupported split {split}. Split must be one of train, val, or test.'
+        self._split = split
+        self._path = os.path.join(path,'yahoo',f'{split}.csv')
+        download(self._INFO[split]['url'],
+                 path=self._path,
+                 sha1_hash=self._INFO[split]['sha1sum'])
+        self._data = pd.read_csv(self._path)
+    
+    @property  
+    def data(self):
+        return self._data
+    
+    @property
+    def label_column(self):
+        return 'target'
+    
+    @property
+    def label_type(self):
+        return NUMERICAL
+    
+    @property
+    def metric(self):
+        return RMSE
+    
+    @property
+    def problem_type(self):
+        return REGRESSION
+
+
+class YearTabularDataset(BaseTabularDataset):
+    _INFO = {
+        'train': {
+            'url': 's3://autogluon/datasets/tabular/year/train.csv',
+            'sha1sum': '1f609cc6fb7d3cb71098041a5082eda190538de8'
+        },
+        'val': {
+            'url': 's3://autogluon/datasets/tabular/year/val.csv',
+            'sha1sum': '4afdc7452aa3ef643d1c24719c91970791c31ecd'
+        },
+        'test': {
+            'url': 's3://autogluon/datasets/tabular/year/test.csv',
+            'sha1sum': 'b412b511519e04444337f82e54e292d6d4beeb5e'
+        },
+    }
+    def __init__(self, split='train', path='./dataset/'):
+        assert split in ['train', 'val', 'test'], f'Unsupported split {split}. Split must be one of train, val, or test.'
+        self._split = split
+        self._path = os.path.join(path,'year',f'{split}.csv')
+        download(self._INFO[split]['url'],
+                 path=self._path,
+                 sha1_hash=self._INFO[split]['sha1sum'])
+        self._data = pd.read_csv(self._path)
+    
+    @property  
+    def data(self):
+        return self._data
+    
+    @property
+    def label_column(self):
+        return 'target'
+    
+    @property
+    def label_type(self):
+        return NUMERICAL
+    
+    @property
+    def metric(self):
+        return RMSE
+    
+    @property
+    def problem_type(self):
+        return REGRESSION
```

Comment:
We may refactor this part later. We can release the auto_mm_bench package or reuse the huggingface datasets API.